drivers/serial/uart_native_pty.c

```diff
@@ -209,6 +248,144 @@ static int np_uart_poll_in(const struct device *dev, unsigned char *p_char)
 	}
 }
 
+#ifdef CONFIG_UART_ASYNC_API
+
+static int np_uart_callback_set(const struct device *dev, uart_callback_t callback, void *user_data)
+{
+	struct native_pty_status *data = dev->data;
+
+	data->async.user_callback = callback;
+	data->async.user_data = user_data;
+
+	return 0;
+}
+
+static void np_uart_tx_done_work(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct native_pty_status *data =
+		CONTAINER_OF(dwork, struct native_pty_status, async.tx_done);
+	struct uart_event evt;
+	unsigned int key = irq_lock();
+	long ret;
+
+	evt.type = UART_TX_DONE;
+	evt.data.tx.buf = data->async.tx_buf;
+	evt.data.tx.len = data->async.tx_len;
+
+	ret = nsi_host_write(data->out_fd, evt.data.tx.buf, evt.data.tx.len);
+	(void)ret;
+
+	data->async.tx_buf = NULL;
+
+	data->async.user_callback(data->async.dev, &evt, data->async.user_data);
+	irq_unlock(key);
+}
+
+static int np_uart_tx(const struct device *dev, const uint8_t *buf, size_t len, int32_t timeout)
+{
+	struct native_pty_status *data = dev->data;
```

Comment:
should this check if the callback is set and return -ENOTSUP otherwise?
(Same for rx and np_uart_tx_abort)

Reply:
That isn't done in the other drivers that I have inspected

Reply:
Ok, I leave it up to the driver class owner @dcpleung to decide if anything needs doing